tools/challenge-parser/parser/plugins/add-seed.js

```diff
@@ -55,26 +55,29 @@ function addSeeds() {
     visitForContents(contentsTree);
     visitForHead(headTree);
     visitForTail(tailTree);
+    const seedVals = Object.values(seeds);
     file.data = {
       ...file.data,
-      files: seeds
+      challengeFiles: seedVals
     };
 
     // process region markers - remove them from content and add them to data
-    Object.keys(seeds).forEach(key => {
-      const fileData = seeds[key];
-      const editRegionMarkers = findRegionMarkers(fileData);
-      if (editRegionMarkers) {
-        fileData.contents = removeLines(fileData.contents, editRegionMarkers);
+    seedVals
+      .map(x => x.fileKey)
+      .forEach(key => {
+        const fileData = seedVals.find(x => x.fileKey === key);
+        const editRegionMarkers = findRegionMarkers(fileData);
+        if (editRegionMarkers) {
+          fileData.contents = removeLines(fileData.contents, editRegionMarkers);
 
-        if (editRegionMarkers[1] <= editRegionMarkers[0]) {
-          throw Error('Editable region must be non zero');
+          if (editRegionMarkers[1] <= editRegionMarkers[0]) {
+            throw Error('Editable region must be non zero');
+          }
+          fileData.editableRegionBoundaries = editRegionMarkers;
+        } else {
+          fileData.editableRegionBoundaries = [];
         }
-        fileData.editableRegionBoundaries = editRegionMarkers;
-      } else {
-        fileData.editableRegionBoundaries = [];
-      }
-    });
+      });
```

Comment:
Since we're cleaning this up, can we make this a bit more functional? It should be a little more obvious what's happening here.
```js
    // process region markers - remove them from content and add them to data
    const challengeFiles = Object.values(seeds).map(data => {
      const seed = { ...data };
      const editRegionMarkers = findRegionMarkers(seed);
      if (editRegionMarkers) {
        seed.contents = removeLines(seed.contents, editRegionMarkers);

        if (editRegionMarkers[1] <= editRegionMarkers[0]) {
          throw Error('Editable region must be non zero');
        }
        seed.editableRegionBoundaries = editRegionMarkers;
      } else {
        seed.editableRegionBoundaries = [];
      }
      return seed;
    });

    file.data = {
      ...file.data,
      challengeFiles
    };
```